src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.Helpers.cs

```diff
@@ -497,7 +497,7 @@ private static NullabilityState DetermineParameterNullability(ParameterInfo para
             {
                 return NullabilityState.NotNull;
             }
-#if NET8_0
+#if NET && !NET9_0_OR_GREATER
```

Comment:
I agree with @jkotas above that this just makes the condition more complex unnecessarily. The comment below says

`// Should be removed once .NET 8 support is dropped.`

so better to just keep this as `NET8_0`.